packages/loader/container-loader/src/protocol.ts

```diff
@@ -83,6 +84,12 @@ export class ProtocolHandler extends ProtocolOpHandler implements IProtocolHandl
             }
         }
 
+        const currentSeqNum = message.clientSequenceNumber;
+        if (client?.lastSequenceNumber !== undefined && !canBeCoalescedByService(message)) {
```

Comment:
Doesn't DeltaManager already have this kind of validation?

https://github.com/vivekmogalapalli/fluidframework/blob/6d121fde3d4f97d8734ae1f1c188ea2e7a62c4df/packages/loader/container-loader/src/deltaManager.ts#L827

What are the cases where this would detect a problem that DeltaManager missed?

I'm not objecting to this additional validation, just curious for how these two paths relate.

Reply:
I do not see existing validation of clientSequenceNumber in DM (the reference above is for latest main, not specific commit :)). There is one for sequenceNumber, yes, but that's different